bin/setup

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+
+##
+# Setup script to symlink configuration and dot files from this repository
+# to proper places in the system.
+#
+# Script is idempotent. It is same to run it several times.
+#
+# TODO:
+
+# Stop immediately if any external program returns a non-zero exit status.
+# Short: set -e
+set -o errexit
+# Treat unset variables as an error when substituting. Short: set -u
+set -o nounset
+# If set, the return value of a pipeline is the value of the last (rightmost)
+# command to exit with a non-zero status, or zero if all commands in the
+# pipeline exit successfully. This option is disabled by default.
+set -o pipefail
+
+# Uncomment for debugging
+#
+# Disable file name generation using metacharacters (globbing).
+# Short: set -f
+#set -o noglob
+# Prints shell input lines as they are read.
+# Short: set -v
+#set -o verbose
+# Print command traces before executing command.
+# Short: set -x
+#set -o xtrace
+
+
+readonly __DIR__="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
Looks like `readonly` is not supported in Bash 3, that means that your script will not work on OS X :+1:

Reply:
I do not plan to support it! But thanks for notice! I didn't know that it is not supported on OS X.

Reply:
I added dependencies version declaration in d304cff5cff22956334b8b271afa8468821156c2.